worlds/tloz/__init__.py

```diff
@@ -0,0 +1,496 @@
+                        elif item_class == ItemClassification.trap:
+                            item_price = item_price * 2
+                    rom_data[price_location] = item_price
+                if location.name == "Take Any Item Right":
+                    # Same story as above: bit 6 is what makes this a Take Any cave
+                    item_id = item_id | 0b01000000
+                rom_data[location_id] = item_id
+
+            # We shuffle the tiers of rupee caves. Caves that shared a value before still will.
+            secret_caves = self.multiworld.random.sample(sorted(secret_money_ids), 3)
+            secret_cave_money_amounts = [20, 50, 100]
+            for i, amount in enumerate(secret_cave_money_amounts):
+                # Giving approximately double the money to keep grinding down
+                amount = amount * self.multiworld.random.triangular(1.5, 2.5)
+                secret_cave_money_amounts[i] = int(amount)
+            for i, cave in enumerate(secret_caves):
+                rom_data[secret_money_ids[cave]] = secret_cave_money_amounts[i]
+            return rom_data
+
+    def generate_output(self, output_directory: str):
+        patched_rom = self.apply_randomizer()
+        outfilebase = 'AP_' + self.multiworld.seed_name
+        outfilepname = f'_P{self.player}'
+        outfilepname += f"_{self.multiworld.get_file_safe_player_name(self.player).replace(' ', '_')}"
+        outputFilename = os.path.join(output_directory, f'{outfilebase}{outfilepname}.nes')
+
+        self.rom_name_text = f'LOZ{Utils.__version__.replace(".", "")[0:3]}_{self.player}_{self.multiworld.seed:11}\0'
+        self.romName = bytearray(self.rom_name_text, 'utf8')[:0x20]
+        self.romName.extend([0] * (0x20 - len(self.romName)))
+        self.rom_name = self.romName
+        patched_rom[0x10:0x30] = self.romName
+
+        self.playerName = bytearray(self.multiworld.player_name[self.player], 'utf8')[:0x20]
+        self.playerName.extend([0] * (0x20 - len(self.playerName)))
+        patched_rom[0x30:0x50] = self.playerName
+
+        patched_filename = os.path.join(output_directory, outputFilename)
+
+        with open(patched_filename, 'wb') as patched_rom_file:
+            patched_rom_file.write(patched_rom)
+
+        patch = TLoZDeltaPatch(os.path.splitext(outputFilename)[0] + TLoZDeltaPatch.patch_file_ending,
+                               player=self.player,
+                               player_name=self.multiworld.player_name[self.player], patched_path=outputFilename)
+        self.rom_name_available_event.set()
+        patch.write()
+        os.unlink(outputFilename)
+
+    def modify_multidata(self, multidata: dict):
+        import base64
+        rom_name = getattr(self, "rom_name", None)
+        if rom_name:
+            new_name = base64.b64encode(bytes(self.rom_name)).decode()
+            multidata["connect_names"][new_name] = multidata["connect_names"][self.multiworld.player_name[self.player]]
```

Comment:
```suggestion
        self.rom_name_available_event.wait()
        new_name = base64.b64encode(bytes(self.rom_name)).decode()
        multidata["connect_names"][new_name] = multidata["connect_names"][self.multiworld.player_name[self.player]]
```

Wait until the other thread has the name, then it it will also exist.